### PR TITLE
Layout editor enhancements

### PR DESCRIPTION
**Pull request by dgutierr**
Video showing both the preview feature plus the layout style selection (fluid vs page) in the page creation popup => https://issues.jboss.org/secure/attachment/12428445/layouteditor-AF911and912.mp4

@ederign @kkufova Mind to review please?

P.S: @pefernan You might get interested in the preview feature. Take also a look please if so.

### CODE REVIEW

**Comment by kkufova** [comment]
@dgutierr, one test failed: `PerspectiveServicesImplTest.testCreate:75 expected:<null> but was:<test>`. Could you please take a look, so I can review?

**Comment by kkufova** [comment]
jenkins execute full downstream build

**Comment by dgutierr** [comment]
@kkufova I had to push another PR to fix the full build => https://github.com/kiegroup/kie-wb-common/pull/1311

Jenkins execute full downstream build

**Comment by kkufova** [comment]
@dgutierr, I'm still getting failures when building `kie-wb-common`:

```
Tests in error: 
  DockerExecutorTest.testAPI:125 » Runtime An error occurred while executing the...
  DockerExecutorTest.testFlexAPI:220 » Runtime An error occurred while executing...
```

**Comment by kkufova** [comment]
jenkins execute full downstream build

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.html [hunk 1]
```diff
@@ -1,25 +1,35 @@
-<div class="le-default">
+<div data-field="mainDiv">
+    <div data-field="tabsDiv">
+        <ul class="nav nav-tabs nav-tabs-pf">
+            <li data-field="designTab" class="active" style="margin-left: 10px"><a href="#" data-field="designAnchor">Editor!</a></li>
+            <li data-field="previewTab"><a href="#" data-field="previewAnchor">Preview!</a></li>
```
Comment (ederign) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.html]: i18n?

Comment (ederign) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.html]: If there is no preview, we will have the design tab right? This is ok for forms? /cc @pefernan

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.java [hunk 2]
```diff
@@ -34,26 +43,90 @@
...
         this.presenter = presenter;
+        componentsLabel.setTextContent(CommonConstants.INSTANCE.Components());
+        designAnchor.setTextContent(CommonConstants.INSTANCE.Editor());
+        previewAnchor.setTextContent(CommonConstants.INSTANCE.Preview());
+    }
+
+    @Override
+    public void setPreviewEnabled(boolean previewEnabled) {
+        tabsDiv.setHidden(!previewEnabled);
+        mainDiv.getStyle().setProperty("height", previewEnabled ? "95%" : "100%");
```
Comment (ederign) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.java]: you hide the tabs here right?

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/generator/AbstractLayoutGenerator.java [hunk 3]
```diff
@@ -17,84 +17,94 @@
...
-import org.uberfire.ext.layout.editor.client.infra.ColumnSizeBuilder;
 import org.uberfire.ext.layout.editor.client.infra.RowSizeBuilder;
 
 public abstract class AbstractLayoutGenerator implements LayoutGenerator {
 
     @Override
-    public Panel build(LayoutTemplate layoutTemplate) {
-        ComplexPanel container = getLayoutContainer();
+    public LayoutInstance build(LayoutTemplate layoutTemplate) {
+        HTMLElement container = createContainer(layoutTemplate);
```
Comment (ederign) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/generator/AbstractLayoutGenerator.java]: Can you please check if we can use elemental API's here?

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.html (hunk 1 above)
Comment (dgutierr) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.html]: @ederign I'm using use GWT i18n to inject constants from the view into the html.
The tabs are disabled by default. They are only explicitly enabled in perspective editor.

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.java (hunk 2 above)
Comment (dgutierr) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorView.java]: @ederign Yep.

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/generator/AbstractLayoutGenerator.java (hunk 3 above)
Comment (dgutierr) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/generator/AbstractLayoutGenerator.java]: @ederign Can you please point me to an example?

**Comment by kkufova** [comment]
@dgutierr, I noticed that the discussed error
> `Unable to complete your request. The following exception occurred: DOMException: WRONG_DOCUMENT_ERR.`

appears when creating a page. I filed a Jira to deal with this: [AF-920](https://issues.jboss.org/browse/AF-920). Can we merge this PR before the problem is solved?

Also, would it make sense to add an option to change the style of the page after the page is already created? WDYT?

Otherwise the changes work well. +1 for me.

**Comment by dgutierr** [comment]
@kkufova The PR can be merged and it should imho.

In my opinion, an user should know in advance what type of page the is aiming for.  IMHO changing the style during edition introduces a few challenges and it doesn't add too much value.

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/generator/AbstractLayoutGenerator.java (hunk 3 above)
Comment (ederign) [reply uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/generator/AbstractLayoutGenerator.java]: @dgutierr @adrielparedes is working on a new feature. @adrielparedes can you please give it an example for @dgutierr ?

@dgutierr next PR's we should all use elemental API's.

**Pull request merged**